src/upgrade/verify.go

```diff
@@ -0,0 +1,111 @@
+package upgrade
+
+import (
+	"crypto"
+	"crypto/rsa"
+	"crypto/sha256"
+	"crypto/x509"
+	_ "embed"
+	"encoding/pem"
+	"fmt"
+	stdruntime "runtime"
+	"strings"
+
+	"github.com/jandedobbeleer/oh-my-posh/src/runtime"
+)
+
+//go:embed public_key.pem
+var publicKey []byte
+
+func downloadAndVerify(tag string) ([]byte, error) {
+	extension := ""
+	if stdruntime.GOOS == runtime.WINDOWS {
+		extension = ".exe"
+	}
+
+	asset := fmt.Sprintf("posh-%s-%s%s", stdruntime.GOOS, stdruntime.GOARCH, extension)
+
+	data, err := downloadReleaseAsset(tag, asset)
+	if err != nil {
+		return nil, err
+	}
+
+	setState(verifying)
+
+	err = verify(tag, asset, data)
+	if err != nil {
+		return nil, err
+	}
+
+	return data, nil
+}
+
+func verify(tag, asset string, binary []byte) error {
+	binaryData, err := downloadReleaseAsset(tag, "checksums.txt")
+	if err != nil {
+		return err
+	}
+
+	signature, err := downloadReleaseAsset(tag, "checksums.txt.sig")
+	if err != nil {
+		return err
+	}
+
+	rsaPublicKey, err := loadPublicKey()
+	if err != nil {
+		return err
+	}
+
+	hashed := sha256.Sum256(binaryData)
+	err = rsa.VerifyPKCS1v15(rsaPublicKey, crypto.SHA256, hashed[:], signature)
```

Comment:
Any reason why not using an ECC algorithm? 

When using RSA, the recommended scheme is RSA-PSS.  
[From the docs](https://pkg.go.dev/crypto/rsa):
> The original specification for encryption and signatures with RSA is PKCS #1 and the terms "RSA encryption" and "RSA signatures" by default refer to PKCS #1 version 1.5. **However, that specification has flaws and new designs should use version 2, usually called by just OAEP and PSS, where possible.**

Reply:
Because I'm not at home in these things, so I need people like you help me out on this one :-) I'll have a look.

Reply:
@kapsiR seems like PSS and `golang` don't play together nicely. I'm out of options as it can't parse on RSA-PSS public key correctly.

Reply:
What about using an ECC algorithm like [Ed25519](https://pkg.go.dev/crypto/ed25519) or [ECDSA](https://pkg.go.dev/crypto/ecdsa)?  
Do you need a PoC?

Reply:
Since Go isn't my main language, I'd give it a try to provide a PoC implementation to gain some experience 😉